examples/qml_features/rust/src/singleton.rs

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+// ANCHOR: book_macro_code
+#[cxx_qt::bridge(cxx_file_stem = "rust_invokables")]
+pub mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qcolor.h");
+        type QColor = cxx_qt_lib::QColor;
+    }
+
+    #[cxx_qt::qobject(qml_uri = "com.kdab.cxx_qt.demo", qml_version = "1.0", qml_singleton)]
```

Comment:
We get `error: 'QML_SINGLETON' does not name a type`, probably a missing include?

Reply:
Yes adding `include!(<QtQml/QQmlEngine>);` makes this work, now we just need it to be injected at the right time

Reply:
I worked around this for the other C++ macros by putting what they expand to into our generated C++, but the [QML_SINGLETON C++ macro](https://codebrowser.dev/qt6/qtdeclarative/src/qml/qml/qqmlregistration.h.html#_M/QML_SINGLETON) is a bit more complicated. I didn't figure out how to get moc to properly handle an `#include <QQml/qqmlregistration.h>` as noted in the [comment](https://github.com/KDAB/cxx-qt/blob/1a4f5aaac1a7714aaa3e711820bd6af74032a9c4/crates/cxx-qt-gen/src/generator/cpp/qobject.rs#L33).

Reply:
Yup, i've got this working locally now, just writing some tests :-) it's simple enough to add `include!(<QtQml/QQmlEngine>);` when we need it.

Reply:
Done

Reply:
Interesting. I don't know why `#include <QQml/qmlregistration.h` didn't work because that's where the C++ macros are defined. I'm wondering if we really need to pull in everything from QQmlEngine. Does `#include <QQml/qqml.h>` work?

Reply:
You probably need the template from here ? https://codebrowser.dev/qt5/qtdeclarative/src/qml/qml/qqmlengine.cpp.html#1399

And the documentation for `QML_SINGLETON` comes from `QQml/QQmlEngine` so that's the best one to include as it's the documented way rather than assuming it'll be in the right headers.

Reply:
> And the documentation for QML_SINGLETON comes from QQml/QQmlEngine so that's the best one to include as it's the documented way rather than assuming it'll be in the right headers.

Okay.